libraries/extractor/src/main/java/androidx/media3/extractor/ts/H265Reader.java

```diff
@@ -435,11 +435,15 @@ private static void skipShortTermRefPicSets(ParsableNalUnitBitArray bitArray) {
       if (interRefPicSetPredictionFlag) {
         bitArray.skipBit(); // delta_rps_sign
         bitArray.readUnsignedExpGolombCodedInt(); // abs_delta_rps_minus1
+        int numDeltaPocs = 0;
         for (int j = 0; j <= previousNumDeltaPocs; j++) {
-          if (bitArray.readBit()) { // used_by_curr_pic_flag[j]
-            bitArray.skipBit(); // use_delta_flag[j]
+          if (!bitArray.readBit()) { // used_by_curr_pic_flag[j]
```

Comment:
This line is clearly fixing the code to be aligned with the spec, and it seems that this change alone makes the provided video play successfully.

I'm afraid I'm not quite following the reasoning about the other changes regarding `numDeltaPocs` - it doesn't seem to reflect my understanding of the pseudocode in section 7.3.7 of the H.265 spec.

I wonder if you'd be up for splitting this into two pull requests - I'm happy to merge this one line as-is (pending my testing question), the rest I want to discuss a bit more.

Reply:
> Do you have a piece of test content you can submit as part of this PR (maybe the video attached to the issue)? It would make sense to use it for a new test method in `TsExtractorTest` and also a new entry in`TsPlaybackTest`. It would give us nice regression protection.

Yes, I have submitted a test video in the issue androidx/media#303, and add the link to "media file".   I have edit that comment, make the link more noticeable.

Reply:
I added a sample ts file to test assets , and added corresponding test method to `TsExtractorTest` and `TsPlaybackTest`.  (
This sample is **not** the one attached in issue #303 . )

If you checkout commit 8f21294fe70a322aded49aecff0c0d07483e0f03 , `TsExtractorTest` would fail as expected.

And I also break the fix into two commits,  292f568 and 3be8729.

* Commit 292f568  only add `!` to condition of `used_by_curr_pic_flag`,  with this commit , `TsExtractorTest` would still fail.
* Commit 3be8729 fixed `numDeltaPocs `, with this commit , `TsExtractorTest` would pass.

In section 7.3.7 of the H.265 spec, `previousNumDeltaPocs` is NumDeltaPocs[ RefRpsIdx ].
And in section 7.4.8 , there are more detail steps of how NumDeltaPocs are calculated. 

PS. Here is another sample code https://github.com/xqq/mpegts.js/blob/master/src/demux/h265-parser.js#L174

Reply:
Thank you for the test asset, I agree that the test only passes with all the changes in this PR. I'm afraid I'm still struggling to resolve the code you've provided against the spec (even looking at section 7.4.8 as well).

I had a go at making it a bit more verbose to match how I think it relates to the spec (i.e. making fewer simplifications/elisions of `7-61` and `7-62`) and I've uploaded that commit to this branch. Can you take a look and see what you think? What's interesting is that it changes (removes) the [Pixel Aspect Ratio (PAR)](https://en.wikipedia.org/wiki/Pixel_aspect_ratio) value from the `Format` that is output by the extractor. I wasn't expecting that.

I analysed the file with `ffprobe` and I **think** the old value (`1.0003651`) wasn't correct, since that seems to be the Storage Aspect Ratio (SAR) according to `ffprobe` (`2740 / 2739 = 1.0003651`):

```shell
$ ffprobe libraries/test_data/src/test/assets/media/ts/sample_h265_rps_pred.ts
<snip>
  Stream #0:0[0x100]: Video: hevc (Main) (HEVC / 0x43564548), yuv420p(tv), 914x686 [SAR 2740:2739 DAR 1045314:784271], 15 fps, 15 tbr, 90k tbn
```

Reply:
> I had a go at making it a bit more verbose to match how I think it relates to the spec (i.e. making fewer simplifications/elisions of `7-61` and `7-62`) and I've uploaded that commit to this branch. Can you take a look and see what you think? What's interesting is that it changes (removes) the [Pixel Aspect Ratio (PAR)](https://en.wikipedia.org/wiki/Pixel_aspect_ratio) value from the `Format` that is output by the extractor. I wasn't expecting that.
> 
> I analysed the file with `ffprobe` and I **think** the old value (`1.0003651`) wasn't correct, since that seems to be the Storage Aspect Ratio (SAR) according to `ffprobe` (`2740 / 2739 = 1.0003651`):
> 
> ```shell
> $ ffprobe libraries/test_data/src/test/assets/media/ts/sample_h265_rps_pred.ts
> <snip>
>   Stream #0:0[0x100]: Video: hevc (Main) (HEVC / 0x43564548), yuv420p(tv), 914x686 [SAR 2740:2739 DAR 1045314:784271], 15 fps, 15 tbr, 90k tbn
> ```
I think `SAR` in ffprob/ffmpeg is `sample aspect ratio`, see [link](https://github.com/FFmpeg/FFmpeg/blob/master/doc/codecs.texi#L344).  

And in function [parseMediaFormat](https://github.com/androidx/media/blob/release/libraries/extractor/src/main/java/androidx/media3/extractor/ts/H265Reader.java#L347).

``` java
            pixelWidthHeightRatio = (float) sarWidth / sarHeight;
```
The value 1.0003651 of  `pixelWidthHeightRatio`  in the dump files are same as SAR from `ffprobe` as expected.

Aspect ratio data are parsed from  SPS  after `ShortTermRefPicSets`, so if `skipShortTermRefPicSets` is not match spec, then following aspect ratio data can't be parsed correctly.

Reply:
I added a new commit to fix your changes. Only when  used_by_curr_pic_flag and use_delta_flag are both 0, numDeltaPocs should not increase.

And maybe you can use trace_headers  bsf in ffmpeg to dump all syntax to help you check it the parsed result from this function match with spec.

```bash
ffmpeg  -i libraries/test_data/src/test/assets/media/ts/sample_h265_rps_pred.ts -c:v copy -bsf filter_units=pass_types=33,trace_headers -f null -
```

<details>
  <summary>SPS syntax  </summary>

# sar (pixelWidthHeightRatio )
```

[trace_headers @ 0x74ce6c0] 627         sar_width                                    0000101010110100 = 2740
[trace_headers @ 0x74ce6c0] 643         sar_height                                   0000101010110011 = 2739
```

# from `short_term_ref_pic_sets`  to the  end
```
[trace_headers @ 0x74ce6c0] 259         num_short_term_ref_pic_sets                         000011101 = 28
[trace_headers @ 0x74ce6c0] 268         num_negative_pics                                       00100 = 3
[trace_headers @ 0x74ce6c0] 273         num_positive_pics                                           1 = 0
[trace_headers @ 0x74ce6c0] 274         delta_poc_s0_minus1[0]                                0001110 = 13
[trace_headers @ 0x74ce6c0] 281         used_by_curr_pic_s0_flag[0]                                 1 = 1
[trace_headers @ 0x74ce6c0] 282         delta_poc_s0_minus1[1]                                  00111 = 6
[trace_headers @ 0x74ce6c0] 287         used_by_curr_pic_s0_flag[1]                                 1 = 1
[trace_headers @ 0x74ce6c0] 288         delta_poc_s0_minus1[2]                                  00111 = 6
[trace_headers @ 0x74ce6c0] 293         used_by_curr_pic_s0_flag[2]                                 1 = 1
[trace_headers @ 0x74ce6c0] 294         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 295         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 296         abs_delta_rps_minus1                                    00111 = 6
[trace_headers @ 0x74ce6c0] 301         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 302         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 303         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 304         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 305         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 306         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 307         abs_delta_rps_minus1                                    00100 = 3
[trace_headers @ 0x74ce6c0] 312         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 313         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 314         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 315         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 316         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 317         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 318         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 319         abs_delta_rps_minus1                                      010 = 1
[trace_headers @ 0x74ce6c0] 322         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 323         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 324         used_by_curr_pic_flag[2]                                    0 = 0
[trace_headers @ 0x74ce6c0] 325         use_delta_flag[2]                                           0 = 0
[trace_headers @ 0x74ce6c0] 326         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 327         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 328         used_by_curr_pic_flag[5]                                    1 = 1
[trace_headers @ 0x74ce6c0] 329         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 330         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 331         abs_delta_rps_minus1                                        1 = 0
[trace_headers @ 0x74ce6c0] 332         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 333         used_by_curr_pic_flag[1]                                    0 = 0
[trace_headers @ 0x74ce6c0] 334         use_delta_flag[1]                                           0 = 0
[trace_headers @ 0x74ce6c0] 335         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 336         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 337         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 338         used_by_curr_pic_flag[5]                                    1 = 1
[trace_headers @ 0x74ce6c0] 339         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 340         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 341         abs_delta_rps_minus1                                      011 = 2
[trace_headers @ 0x74ce6c0] 344         used_by_curr_pic_flag[0]                                    0 = 0
[trace_headers @ 0x74ce6c0] 345         use_delta_flag[0]                                           0 = 0
[trace_headers @ 0x74ce6c0] 346         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 347         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 348         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 349         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 350         used_by_curr_pic_flag[5]                                    0 = 0
[trace_headers @ 0x74ce6c0] 351         use_delta_flag[5]                                           0 = 0
[trace_headers @ 0x74ce6c0] 352         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 353         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 354         abs_delta_rps_minus1                                        1 = 0
[trace_headers @ 0x74ce6c0] 355         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 356         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 357         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 358         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 359         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 360         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 361         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 362         abs_delta_rps_minus1                                      010 = 1
[trace_headers @ 0x74ce6c0] 365         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 366         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 367         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 368         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 369         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 370         used_by_curr_pic_flag[5]                                    0 = 0
[trace_headers @ 0x74ce6c0] 371         use_delta_flag[5]                                           0 = 0
[trace_headers @ 0x74ce6c0] 372         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 373         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 374         abs_delta_rps_minus1                                    00100 = 3
[trace_headers @ 0x74ce6c0] 379         used_by_curr_pic_flag[0]                                    0 = 0
[trace_headers @ 0x74ce6c0] 380         use_delta_flag[0]                                           0 = 0
[trace_headers @ 0x74ce6c0] 381         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 382         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 383         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 384         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 385         used_by_curr_pic_flag[5]                                    0 = 0
[trace_headers @ 0x74ce6c0] 386         use_delta_flag[5]                                           0 = 0
[trace_headers @ 0x74ce6c0] 387         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 388         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 389         abs_delta_rps_minus1                                      010 = 1
[trace_headers @ 0x74ce6c0] 392         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 393         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 394         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 395         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 396         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 397         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 398         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 399         abs_delta_rps_minus1                                        1 = 0
[trace_headers @ 0x74ce6c0] 400         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 401         used_by_curr_pic_flag[1]                                    0 = 0
[trace_headers @ 0x74ce6c0] 402         use_delta_flag[1]                                           0 = 0
[trace_headers @ 0x74ce6c0] 403         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 404         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 405         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 406         used_by_curr_pic_flag[5]                                    1 = 1
[trace_headers @ 0x74ce6c0] 407         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 408         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 409         abs_delta_rps_minus1                                      011 = 2
[trace_headers @ 0x74ce6c0] 412         used_by_curr_pic_flag[0]                                    0 = 0
[trace_headers @ 0x74ce6c0] 413         use_delta_flag[0]                                           0 = 0
[trace_headers @ 0x74ce6c0] 414         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 415         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 416         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 417         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 418         used_by_curr_pic_flag[5]                                    0 = 0
[trace_headers @ 0x74ce6c0] 419         use_delta_flag[5]                                           0 = 0
[trace_headers @ 0x74ce6c0] 420         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 421         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 422         abs_delta_rps_minus1                                        1 = 0
[trace_headers @ 0x74ce6c0] 423         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 424         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 425         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 426         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 427         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 428         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 429         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 430         abs_delta_rps_minus1                                      010 = 1
[trace_headers @ 0x74ce6c0] 433         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 434         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 435         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 436         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 437         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 438         used_by_curr_pic_flag[5]                                    0 = 0
[trace_headers @ 0x74ce6c0] 439         use_delta_flag[5]                                           0 = 0
[trace_headers @ 0x74ce6c0] 440         inter_ref_pic_set_prediction_flag                           0 = 0
[trace_headers @ 0x74ce6c0] 441         num_negative_pics                                       00100 = 3
[trace_headers @ 0x74ce6c0] 446         num_positive_pics                                           1 = 0
[trace_headers @ 0x74ce6c0] 447         delta_poc_s0_minus1[0]                                0001110 = 13
[trace_headers @ 0x74ce6c0] 454         used_by_curr_pic_s0_flag[0]                                 1 = 1
[trace_headers @ 0x74ce6c0] 455         delta_poc_s0_minus1[1]                                  00111 = 6
[trace_headers @ 0x74ce6c0] 460         used_by_curr_pic_s0_flag[1]                                 1 = 1
[trace_headers @ 0x74ce6c0] 461         delta_poc_s0_minus1[2]                                  00111 = 6
[trace_headers @ 0x74ce6c0] 466         used_by_curr_pic_s0_flag[2]                                 1 = 1
[trace_headers @ 0x74ce6c0] 467         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 468         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 469         abs_delta_rps_minus1                                    00111 = 6
[trace_headers @ 0x74ce6c0] 474         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 475         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 476         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 477         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 478         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 479         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 480         abs_delta_rps_minus1                                    00100 = 3
[trace_headers @ 0x74ce6c0] 485         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 486         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 487         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 488         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 489         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 490         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 491         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 492         abs_delta_rps_minus1                                      010 = 1
[trace_headers @ 0x74ce6c0] 495         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 496         used_by_curr_pic_flag[1]                                    0 = 0
[trace_headers @ 0x74ce6c0] 497         use_delta_flag[1]                                           0 = 0
[trace_headers @ 0x74ce6c0] 498         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 499         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 500         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 501         used_by_curr_pic_flag[5]                                    1 = 1
[trace_headers @ 0x74ce6c0] 502         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 503         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 504         abs_delta_rps_minus1                                        1 = 0
[trace_headers @ 0x74ce6c0] 505         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 506         used_by_curr_pic_flag[1]                                    0 = 0
[trace_headers @ 0x74ce6c0] 507         use_delta_flag[1]                                           0 = 0
[trace_headers @ 0x74ce6c0] 508         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 509         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 510         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 511         used_by_curr_pic_flag[5]                                    1 = 1
[trace_headers @ 0x74ce6c0] 512         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 513         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 514         abs_delta_rps_minus1                                      011 = 2
[trace_headers @ 0x74ce6c0] 517         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 518         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 519         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 520         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 521         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 522         used_by_curr_pic_flag[5]                                    0 = 0
[trace_headers @ 0x74ce6c0] 523         use_delta_flag[5]                                           0 = 0
[trace_headers @ 0x74ce6c0] 524         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 525         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 526         abs_delta_rps_minus1                                        1 = 0
[trace_headers @ 0x74ce6c0] 527         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 528         used_by_curr_pic_flag[1]                                    0 = 0
[trace_headers @ 0x74ce6c0] 529         use_delta_flag[1]                                           0 = 0
[trace_headers @ 0x74ce6c0] 530         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 531         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 532         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 533         used_by_curr_pic_flag[5]                                    1 = 1
[trace_headers @ 0x74ce6c0] 534         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 535         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 536         abs_delta_rps_minus1                                      010 = 1
[trace_headers @ 0x74ce6c0] 539         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 540         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 541         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 542         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 543         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 544         used_by_curr_pic_flag[5]                                    0 = 0
[trace_headers @ 0x74ce6c0] 545         use_delta_flag[5]                                           0 = 0
[trace_headers @ 0x74ce6c0] 546         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 547         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 548         abs_delta_rps_minus1                                    00100 = 3
[trace_headers @ 0x74ce6c0] 553         used_by_curr_pic_flag[0]                                    0 = 0
[trace_headers @ 0x74ce6c0] 554         use_delta_flag[0]                                           0 = 0
[trace_headers @ 0x74ce6c0] 555         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 556         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 557         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 558         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 559         used_by_curr_pic_flag[5]                                    0 = 0
[trace_headers @ 0x74ce6c0] 560         use_delta_flag[5]                                           0 = 0
[trace_headers @ 0x74ce6c0] 561         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 562         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 563         abs_delta_rps_minus1                                      010 = 1
[trace_headers @ 0x74ce6c0] 566         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 567         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 568         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 569         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 570         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 571         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 572         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 573         abs_delta_rps_minus1                                        1 = 0
[trace_headers @ 0x74ce6c0] 574         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 575         used_by_curr_pic_flag[1]                                    0 = 0
[trace_headers @ 0x74ce6c0] 576         use_delta_flag[1]                                           0 = 0
[trace_headers @ 0x74ce6c0] 577         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 578         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 579         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 580         used_by_curr_pic_flag[5]                                    1 = 1
[trace_headers @ 0x74ce6c0] 581         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 582         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 583         abs_delta_rps_minus1                                      011 = 2
[trace_headers @ 0x74ce6c0] 586         used_by_curr_pic_flag[0]                                    0 = 0
[trace_headers @ 0x74ce6c0] 587         use_delta_flag[0]                                           0 = 0
[trace_headers @ 0x74ce6c0] 588         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 589         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 590         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 591         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 592         used_by_curr_pic_flag[5]                                    0 = 0
[trace_headers @ 0x74ce6c0] 593         use_delta_flag[5]                                           0 = 0
[trace_headers @ 0x74ce6c0] 594         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 595         delta_rps_sign                                              0 = 0
[trace_headers @ 0x74ce6c0] 596         abs_delta_rps_minus1                                        1 = 0
[trace_headers @ 0x74ce6c0] 597         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 598         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 599         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 600         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 601         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 602         inter_ref_pic_set_prediction_flag                           1 = 1
[trace_headers @ 0x74ce6c0] 603         delta_rps_sign                                              1 = 1
[trace_headers @ 0x74ce6c0] 604         abs_delta_rps_minus1                                      010 = 1
[trace_headers @ 0x74ce6c0] 607         used_by_curr_pic_flag[0]                                    1 = 1
[trace_headers @ 0x74ce6c0] 608         used_by_curr_pic_flag[1]                                    1 = 1
[trace_headers @ 0x74ce6c0] 609         used_by_curr_pic_flag[2]                                    1 = 1
[trace_headers @ 0x74ce6c0] 610         used_by_curr_pic_flag[3]                                    1 = 1
[trace_headers @ 0x74ce6c0] 611         used_by_curr_pic_flag[4]                                    1 = 1
[trace_headers @ 0x74ce6c0] 612         used_by_curr_pic_flag[5]                                    0 = 0
[trace_headers @ 0x74ce6c0] 613         use_delta_flag[5]                                           0 = 0
[trace_headers @ 0x74ce6c0] 614         long_term_ref_pics_present_flag                             0 = 0
[trace_headers @ 0x74ce6c0] 615         sps_temporal_mvp_enabled_flag                               1 = 1
[trace_headers @ 0x74ce6c0] 616         strong_intra_smoothing_enabled_flag                         1 = 1
[trace_headers @ 0x74ce6c0] 617         vui_parameters_present_flag                                 1 = 1
[trace_headers @ 0x74ce6c0] 618         aspect_ratio_info_present_flag                              1 = 1
[trace_headers @ 0x74ce6c0] 619         aspect_ratio_idc                                     11111111 = 255
[trace_headers @ 0x74ce6c0] 627         sar_width                                    0000101010110100 = 2740
[trace_headers @ 0x74ce6c0] 643         sar_height                                   0000101010110011 = 2739
[trace_headers @ 0x74ce6c0] 659         overscan_info_present_flag                                  0 = 0
[trace_headers @ 0x74ce6c0] 660         video_signal_type_present_flag                              0 = 0
[trace_headers @ 0x74ce6c0] 661         chroma_loc_info_present_flag                                0 = 0
[trace_headers @ 0x74ce6c0] 662         neutral_chroma_indication_flag                              0 = 0
[trace_headers @ 0x74ce6c0] 663         field_seq_flag                                              0 = 0
[trace_headers @ 0x74ce6c0] 664         frame_field_info_present_flag                               0 = 0
[trace_headers @ 0x74ce6c0] 665         default_display_window_flag                                 0 = 0
[trace_headers @ 0x74ce6c0] 666         vui_timing_info_present_flag                                1 = 1
[trace_headers @ 0x74ce6c0] 667         vui_num_units_in_tick        00000000000000000000001111101000 = 1000
[trace_headers @ 0x74ce6c0] 699         vui_time_scale               00000000000000000011101010011000 = 15000
[trace_headers @ 0x74ce6c0] 731         vui_poc_proportional_to_timing_flag                         0 = 0
[trace_headers @ 0x74ce6c0] 732         vui_hrd_parameters_present_flag                             0 = 0
[trace_headers @ 0x74ce6c0] 733         bitstream_restriction_flag                                  0 = 0
[trace_headers @ 0x74ce6c0] 734         sps_extension_present_flag                                  0 = 0
[trace_headers @ 0x74ce6c0] 735         rbsp_stop_one_bit                                           1 = 1
```

</details>